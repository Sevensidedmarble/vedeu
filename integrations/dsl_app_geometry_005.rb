#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = 'dsl_app_geometry_005'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    log Dir.tmpdir + '/vedeu.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename:   Dir.tmpdir + "/#{TESTCASE}.out",
                  write_file: true)
              ]
    run_once!
    standalone!

    height 25
    width  80
  end

  Vedeu.interface :top_left_view do
    border do
      foreground '#f44336'
      title 'Top Left'
    end
    geometry do
      align vertical: :top, horizontal: :left, width: 20, height: 4
    end
  end

  Vedeu.interface :top_centre_view do
    border do
      foreground '#e91e63'
      title 'Top Centre'
    end
    geometry do
      align vertical: :top, horizontal: :centre, width: 20, height: 4
    end
  end

  Vedeu.interface :top_right_view do
    border do
      foreground '#9c27b0'
      title 'Top Right'
    end
    geometry do
      align vertical: :top, horizontal: :right, width: 20, height: 4
    end
  end

  Vedeu.interface :middle_left_view do
    border do
      foreground '#673ab7'
      title 'Middle Left'
    end
    geometry do
      align vertical: :middle, horizontal: :left, width: 20, height: 4
    end
  end

  Vedeu.interface :middle_centre_view do
    border do
      foreground '#3f51b5'
      title 'Middle Centre'
    end
    geometry do
      align vertical: :middle, horizontal: :centre, width: 20, height: 4
    end
  end

  Vedeu.interface :middle_right_view do
    border do
      foreground '#2196f3'
      title 'Middle Right'
    end
    geometry do
      align vertical: :middle, horizontal: :right, width: 20, height: 4
    end
  end

  Vedeu.interface :bottom_left_view do
    border do
      foreground '#03a9f4'
      title 'Bottom Left'
    end
    geometry do
      align vertical: :bottom, horizontal: :left, width: 20, height: 4
    end
  end

  Vedeu.interface :bottom_centre_view do
    border do
      foreground '#00bcd4'
      title 'Bottom Centre'
    end
    geometry do
      align vertical: :bottom, horizontal: :centre, width: 20, height: 4
    end
  end

  Vedeu.interface :bottom_right_view do
    border do
      foreground '#009688'
      title 'Bottom Right'
    end
    geometry do
      align vertical: :bottom, horizontal: :right, width: 20, height: 4
    end
  end

  Vedeu.render do
    view(:top_left_view) do
      lines do
        line "top left"
      end
    end

    view(:top_centre_view) do
      lines do
        line "top centre", align: :centre
      end
    end

    view(:top_right_view) do
      lines do
        line "top right", align: :right
      end
    end

    view(:middle_left_view) do
      lines do
        line "middle left"
      end
    end

    view(:middle_centre_view) do
      lines do
        line "middle centre", align: :centre
      end
    end

    view(:middle_right_view) do
      lines do
        line "middle right", align: :right
      end
    end

    view(:bottom_left_view) do
      lines do
        line "bottom left"
      end
    end

    view(:bottom_centre_view) do
      lines do
        line "bottom centre", align: :centre
      end
    end

    view(:bottom_right_view) do
      lines do
        line "bottom right", align: :right
      end
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

Vedeu.timer('Test') do
  DSLApp.start
end

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
