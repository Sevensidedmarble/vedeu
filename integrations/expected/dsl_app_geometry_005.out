[0m[1;1H[39m[49m(0l(B[1;2H(0q(B[1;3H [1;4HT[1;5Ho[1;6Hp[1;7H [1;8HL[1;9He[1;10Hf[1;11Ht[1;12H [1;13H(0q(B[1;14H(0q(B[1;15H(0q(B[1;16H(0q(B[1;17H(0q(B[1;18H(0q(B[1;19H(0q(B[1;20H(0k(B[1;30H(0l(B[1;31H(0q(B[1;32H [1;33HT[1;34Ho[1;35Hp[1;36H [1;37HC[1;38He[1;39Hn[1;40Ht[1;41Hr[1;42He[1;43H [1;44H(0q(B[1;45H(0q(B[1;46H(0q(B[1;47H(0q(B[1;48H(0q(B[1;49H(0q(B[1;50H(0k(B[1;60H(0l(B[1;61H(0q(B[1;62H [1;63HT[1;64Ho[1;65Hp[1;66H [1;67HR[1;68Hi[1;69Hg[1;70Hh[1;71Ht[1;72H [1;73H(0q(B[1;74H(0q(B[1;75H(0q(B[1;76H(0q(B[1;77H(0q(B[1;78H(0q(B[1;79H(0q(B[1;80H(0k(B[0m[2;1H[39m[49m(0x(B[2;2Ht[2;3Ho[2;4Hp[2;5H [2;6Hl[2;7He[2;8Hf[2;9Ht[2;20H(0x(B[2;30H(0x(B[2;31Ht[2;32Ho[2;33Hp[2;34H [2;35Hc[2;36He[2;37Hn[2;38Ht[2;39Hr[2;40He[2;50H(0x(B[2;60H(0x(B[2;61Ht[2;62Ho[2;63Hp[2;64H [2;65Hr[2;66Hi[2;67Hg[2;68Hh[2;69Ht[2;80H(0x(B[0m[3;1H[39m[49m(0x(B[3;20H(0x(B[3;30H(0x(B[3;50H(0x(B[3;60H(0x(B[3;80H(0x(B[0m[4;1H[39m[49m(0m(B[4;2H(0q(B[4;3H(0q(B[4;4H(0q(B[4;5H(0q(B[4;6H(0q(B[4;7H(0q(B[4;8H(0q(B[4;9H(0q(B[4;10H(0q(B[4;11H(0q(B[4;12H(0q(B[4;13H(0q(B[4;14H(0q(B[4;15H(0q(B[4;16H(0q(B[4;17H(0q(B[4;18H(0q(B[4;19H(0q(B[4;20H(0j(B[4;30H(0m(B[4;31H(0q(B[4;32H(0q(B[4;33H(0q(B[4;34H(0q(B[4;35H(0q(B[4;36H(0q(B[4;37H(0q(B[4;38H(0q(B[4;39H(0q(B[4;40H(0q(B[4;41H(0q(B[4;42H(0q(B[4;43H(0q(B[4;44H(0q(B[4;45H(0q(B[4;46H(0q(B[4;47H(0q(B[4;48H(0q(B[4;49H(0q(B[4;50H(0j(B[4;60H(0m(B[4;61H(0q(B[4;62H(0q(B[4;63H(0q(B[4;64H(0q(B[4;65H(0q(B[4;66H(0q(B[4;67H(0q(B[4;68H(0q(B[4;69H(0q(B[4;70H(0q(B[4;71H(0q(B[4;72H(0q(B[4;73H(0q(B[4;74H(0q(B[4;75H(0q(B[4;76H(0q(B[4;77H(0q(B[4;78H(0q(B[4;79H(0q(B[4;80H(0j(B[0m[10;1H[39m[49m(0l(B[10;2H(0q(B[10;3H [10;4HM[10;5Hi[10;6Hd[10;7Hd[10;8Hl[10;9He[10;10H [10;11HL[10;12He[10;13Hf[10;14Ht[10;15H [10;16H(0q(B[10;17H(0q(B[10;18H(0q(B[10;19H(0q(B[10;20H(0k(B[10;30H(0l(B[10;31H(0q(B[10;32H [10;33HM[10;34Hi[10;35Hd[10;36Hd[10;37Hl[10;38He[10;39H [10;40HC[10;41He[10;42Hn[10;43Ht[10;44Hr[10;45He[10;46H [10;47H(0q(B[10;48H(0q(B[10;49H(0q(B[10;50H(0k(B[10;60H(0l(B[10;61H(0q(B[10;62H [10;63HM[10;64Hi[10;65Hd[10;66Hd[10;67Hl[10;68He[10;69H [10;70HR[10;71Hi[10;72Hg[10;73Hh[10;74Ht[10;75H [10;76H(0q(B[10;77H(0q(B[10;78H(0q(B[10;79H(0q(B[10;80H(0k(B[0m[11;1H[39m[49m(0x(B[11;2Hm[11;3Hi[11;4Hd[11;5Hd[11;6Hl[11;7He[11;8H [11;9Hl[11;10He[11;11Hf[11;12Ht[11;20H(0x(B[11;30H(0x(B[11;31Hm[11;32Hi[11;33Hd[11;34Hd[11;35Hl[11;36He[11;37H [11;38Hc[11;39He[11;40Hn[11;41Ht[11;42Hr[11;43He[11;50H(0x(B[11;60H(0x(B[11;61Hm[11;62Hi[11;63Hd[11;64Hd[11;65Hl[11;66He[11;67H [11;68Hr[11;69Hi[11;70Hg[11;71Hh[11;72Ht[11;80H(0x(B[0m[12;1H[39m[49m(0x(B[12;20H(0x(B[12;30H(0x(B[12;50H(0x(B[12;60H(0x(B[12;80H(0x(B[0m[13;1H[39m[49m(0x(B[13;20H(0x(B[13;30H(0x(B[13;50H(0x(B[13;60H(0x(B[13;80H(0x(B[0m[14;1H[39m[49m(0m(B[14;2H(0q(B[14;3H(0q(B[14;4H(0q(B[14;5H(0q(B[14;6H(0q(B[14;7H(0q(B[14;8H(0q(B[14;9H(0q(B[14;10H(0q(B[14;11H(0q(B[14;12H(0q(B[14;13H(0q(B[14;14H(0q(B[14;15H(0q(B[14;16H(0q(B[14;17H(0q(B[14;18H(0q(B[14;19H(0q(B[14;20H(0j(B[14;30H(0m(B[14;31H(0q(B[14;32H(0q(B[14;33H(0q(B[14;34H(0q(B[14;35H(0q(B[14;36H(0q(B[14;37H(0q(B[14;38H(0q(B[14;39H(0q(B[14;40H(0q(B[14;41H(0q(B[14;42H(0q(B[14;43H(0q(B[14;44H(0q(B[14;45H(0q(B[14;46H(0q(B[14;47H(0q(B[14;48H(0q(B[14;49H(0q(B[14;50H(0j(B[14;60H(0m(B[14;61H(0q(B[14;62H(0q(B[14;63H(0q(B[14;64H(0q(B[14;65H(0q(B[14;66H(0q(B[14;67H(0q(B[14;68H(0q(B[14;69H(0q(B[14;70H(0q(B[14;71H(0q(B[14;72H(0q(B[14;73H(0q(B[14;74H(0q(B[14;75H(0q(B[14;76H(0q(B[14;77H(0q(B[14;78H(0q(B[14;79H(0q(B[14;80H(0j(B[0m[21;1H[39m[49m(0l(B[21;2H(0q(B[21;3H [21;4HB[21;5Ho[21;6Ht[21;7Ht[21;8Ho[21;9Hm[21;10H [21;11HL[21;12He[21;13Hf[21;14Ht[21;15H [21;16H(0q(B[21;17H(0q(B[21;18H(0q(B[21;19H(0q(B[21;20H(0k(B[21;30H(0l(B[21;31H(0q(B[21;32H [21;33HB[21;34Ho[21;35Ht[21;36Ht[21;37Ho[21;38Hm[21;39H [21;40HC[21;41He[21;42Hn[21;43Ht[21;44Hr[21;45He[21;46H [21;47H(0q(B[21;48H(0q(B[21;49H(0q(B[21;50H(0k(B[21;60H(0l(B[21;61H(0q(B[21;62H [21;63HB[21;64Ho[21;65Ht[21;66Ht[21;67Ho[21;68Hm[21;69H [21;70HR[21;71Hi[21;72Hg[21;73Hh[21;74Ht[21;75H [21;76H(0q(B[21;77H(0q(B[21;78H(0q(B[21;79H(0q(B[21;80H(0k(B[0m[22;1H[39m[49m(0x(B[22;2Hb[22;3Ho[22;4Ht[22;5Ht[22;6Ho[22;7Hm[22;8H [22;9Hl[22;10He[22;11Hf[22;12Ht[22;20H(0x(B[22;30H(0x(B[22;31Hb[22;32Ho[22;33Ht[22;34Ht[22;35Ho[22;36Hm[22;37H [22;38Hc[22;39He[22;40Hn[22;41Ht[22;42Hr[22;43He[22;50H(0x(B[22;60H(0x(B[22;61Hb[22;62Ho[22;63Ht[22;64Ht[22;65Ho[22;66Hm[22;67H [22;68Hr[22;69Hi[22;70Hg[22;71Hh[22;72Ht[22;80H(0x(B[0m[23;1H[39m[49m(0x(B[23;20H(0x(B[23;30H(0x(B[23;50H(0x(B[23;60H(0x(B[23;80H(0x(B[0m[24;1H[39m[49m(0x(B[24;20H(0x(B[24;30H(0x(B[24;50H(0x(B[24;60H(0x(B[24;80H(0x(B[0m[25;1H[39m[49m(0m(B[25;2H(0q(B[25;3H(0q(B[25;4H(0q(B[25;5H(0q(B[25;6H(0q(B[25;7H(0q(B[25;8H(0q(B[25;9H(0q(B[25;10H(0q(B[25;11H(0q(B[25;12H(0q(B[25;13H(0q(B[25;14H(0q(B[25;15H(0q(B[25;16H(0q(B[25;17H(0q(B[25;18H(0q(B[25;19H(0q(B[25;20H(0j(B[25;30H(0m(B[25;31H(0q(B[25;32H(0q(B[25;33H(0q(B[25;34H(0q(B[25;35H(0q(B[25;36H(0q(B[25;37H(0q(B[25;38H(0q(B[25;39H(0q(B[25;40H(0q(B[25;41H(0q(B[25;42H(0q(B[25;43H(0q(B[25;44H(0q(B[25;45H(0q(B[25;46H(0q(B[25;47H(0q(B[25;48H(0q(B[25;49H(0q(B[25;50H(0j(B[25;60H(0m(B[25;61H(0q(B[25;62H(0q(B[25;63H(0q(B[25;64H(0q(B[25;65H(0q(B[25;66H(0q(B[25;67H(0q(B[25;68H(0q(B[25;69H(0q(B[25;70H(0q(B[25;71H(0q(B[25;72H(0q(B[25;73H(0q(B[25;74H(0q(B[25;75H(0q(B[25;76H(0q(B[25;77H(0q(B[25;78H(0q(B[25;79H(0q(B[25;80H(0j(B[0m